core/src/main/scala/caliban/execution/Executor.scala

```diff
@@ -126,10 +126,10 @@ object Executor {
         }
     }
 
-    for {
-      cache    <- makeCache
-      response <- runQuery(stepReducer.reduceStep(plan, request.field, Map.empty, Nil), cache)
-    } yield response
+    stepReducer.reduceStep(plan, request.field, Map.empty, Nil) match {
+      case PureStep(resp) => Exit.succeed(GraphQLResponse(resp, Nil))
+      case reducedStep    => makeCache.flatMap(runQuery(reducedStep, _))
+    }
```

Comment:
Don't we still need to suspend this? Returning an Exit means we lose referential transparency if it's now getting strictly evaluated.